pkg/pillar/base/logobjecttypes.go

```diff
@@ -0,0 +1,443 @@
+// Copyright (c) 2020 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package base
+
+import (
+	"encoding/json"
+	"github.com/satori/go.uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+// ObjectType : Predefined object types
+type ObjectType string
+
+// The following is NOT an extensive list.
+// Intent is to create a reference list for engineers to get an idea of its usage.
+const (
+	UnknownType  ObjectType = ""
+	LogType      ObjectType = "log"
+	RelationType ObjectType = "relation"
+)
+
+// LogObjectType :
+type LogObjectType string
+
+const (
+	// UnknownLogType : Invalid log type
+	UnknownLogType LogObjectType = ""
+	// ImageLogType :
+	ImageLogType LogObjectType = "image"
+	// NetworkInstanceLogType :
+	NetworkInstanceLogType LogObjectType = "network_instance"
+	// AppInstanceStatusLogType :
+	AppInstanceStatusLogType LogObjectType = "app_instance_status"
+	// AppInstanceConfigLogType :
+	AppInstanceConfigLogType LogObjectType = "app_instance_config"
+	// VolumeLogType :
+	VolumeLogType LogObjectType = "volume_type"
+)
+
+// RelationObjectType :
+type RelationObjectType string
+
+const (
+	// UnknownRelationType : Invalid relation type
+	UnknownRelationType RelationObjectType = ""
+	// AddRelationType :
+	AddRelationType RelationObjectType = "add_relation"
+	// DeleteRelationType :
+	DeleteRelationType RelationObjectType = "delete_relation"
+)
+
+// LogObject : Holds all key value pairs to be logged later.
+type LogObject struct {
+	Initialized bool
+	Fields      map[string]interface{}
+}
+
+var logObjectMap = make(map[string]*LogObject)
+
+// LoggableObject :
+type LoggableObject interface {
+	LogKey() string
+	LogCreate()
+	LogModify(old interface{})
+	LogDelete()
+}
+
+// NewLogObject :
+// objType -> [MANDATORY] log, action, relation, app_instance, image etc
+// objName -> App instance name, name of file being downloaded etc
+// objUUID -> UUID of the object if present (App instance UUID) or UUID of parent object
```

Comment:
I don't think we don't know the parent object with a UUID in all cases.
In any case, it would be confusing to sometimes have the UUID reference to self and in other cases to parent.
If we want a UUID we should have caller pass the zero/uninitialized one and have this function not include it if zero.

Reply:
will do